### ISSUE TITLE
[TW-891] SEGSEV 1.9.4 task log

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#926]
_Bryce Harrington on 2011-01-26T15:47:20Z says:_ 

 blumonc:~/ubuntu/xserver-xorg-video-intel/xserver-xorg-video-intel-2.13.901/debian/patches$ task log pro:x Finalize and upload -intel 2.14.0
Segmentation fault
blumonc:~/ubuntu/xserver-xorg-video-intel/xserver-xorg-video-intel-2.13.901/debian/patches$ gdb task
GNU gdb (GDB) 7.2-ubuntu
Copyright (C) 2010 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "i686-linux-gnu".
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>...
Reading symbols from /usr/local/bin/task...done.
(gdb) run log pro:x Finalize and upload -intel 2.14.0
Starting program: /usr/local/bin/task log pro:x Finalize and upload -intel 2.14.0
[Thread debugging using libthread_db enabled]

Program received signal SIGSEGV, Segmentation fault.
0x0030f68d in fseek () from /lib/libc.so.6
(gdb) bt full
1. 0  0x0030f68d in fseek () from /lib/libc.so.6
No symbol table info available.
1. 1  0x0809c3b6 in TDB::loadCompleted (this=0x81819a8, tasks=..., filter=...) at TDB.cpp:394
        line = '\000' <repeats 30024 times>"\364, \177@\000\b\341\377\277\270\341\377\277\t\000\000\000\326\030/\000\260\341\377\277\260\341\377\277\002\000\000\000\n\000\000\000\260\341\377\277?-\000\375./\000\021\341\377\277gj>\000\001\000\000\000X\341\377\277\340\347\377\277\330\342\377\277,./\000\060.000068", '\000' <repeats 32 times>, "0\000\000\000.\000\000\000\060\000\000\000\060\000\000\000\060\000\000\000\060\000\000\000\066\000\000\000\070\000\000\000\001", '\000' <repeats 55 times>"\340, \221\231\231\231\231\231\231\r\000\000\000\000\000\000\000g\323\021?\260\377\377\377\340\347\377\277\330\342\377\277\200\351\360\377\377\377\377\377\t\000\000\000\224\342\377\277\274\342\377\277fL\301\032g\323\021?", '\000' <repeats 52 times>, ".\000\000\000g\323\021?", '\000' <repeats 12 times>"\364, \177@\000T\341\377\277\000\000\000\000\t\000\000f\377\377\377\177\000\000\000\000\060\341\377\277\b\000\000\000\370\377\377\377\000\000\000\000\070\341\377\277\000\000\000\000\340\347\377\277\000\000\000\000\000\000\000\000gj>"...
        t = {mDescription = {static npos = <optimized out>, 
            _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>}, _M_p = 0x819073c "TDB::loadCompleted"`, mStart = {tv_sec = 1296074661, tv_usec = 598794`
        file = {static npos = <optimized out>, 
          _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>}, _M_p = 0x8195b3c "/home/bryce/.task/completed.data"`
        line_number = 1
1. 2  0x080a1984 in TDB::load (this=0x81819a8, tasks=..., filter=...) at TDB.cpp:271
        numberStatusClauses = 7
        numberSimpleStatusClauses = 135856544
1. 3  0x08123a21 in onProjectChange (task=..., scope=true) at report.cpp:2884
        count_pending = 0
        count_done = 0
        all = {<std::_Vector_base<Task, std::allocator<Task> >> = {
            _M_impl = {<std::allocator<Task>> = {<__gnu_cxx::new_allocator<Task>> = {<No data fields>}, <No data fields>}, _M_start = 0x81902c0, _M_finish = 0x81902e0, _M_end_of_storage = 0x81902e0`, <No data fields>}
        filter = {<std::vector<Att, std::allocator<Att> >> = {<std::_Vector_base<Att, std::allocator<Att> >> = {
              _M_impl = {<std::allocator<Att>> = {<__gnu_cxx::new_allocator<Att>> = {<No data fields>}, <No data fields---Type <return> to continue, or q <return> to quit---
>}, _M_start = 0x0, _M_finish = 0x0, _M_end_of_storage = 0x0`, <No data fields>}, <No data fields>}
        percentage = 0
        msg = <incomplete type>
        project = {static npos = <optimized out>, 
          _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>}, _M_p = 0x818f2e4 "x"`
1. 4  0x080c84d0 in handleLog (outs=...) at command.cpp:203
        out = <incomplete type>
        entryTime = "1296074661\000\b\370\356\377\277"
1. 5  0x0806ba60 in Context::dispatch (this=0x8181940, out=...) at Context.cpp:225
        rc = -1073746176
        t = {mDescription = {static npos = <optimized out>, 
            _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>}, _M_p = 0x81907dc "Context::dispatch"`, mStart = {tv_sec = 1296074661, tv_usec = 598539`
1. 6  0x0806c30d in Context::run (this=0x8181940) at Context.cpp:144
        rc = -1073745976
        output = {static npos = <optimized out>, 
          _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>}, _M_p = 0x81817cc ""`
1. 7  0x08110606 in main (argc=8, argv=0xbffff174) at main.cpp:88
        itask = 4294967295
        tv = {tv_sec = 1296074661, tv_usec = 590594}
        status = <value optimized out>
(gdb) 

[Setting to priority high since it's a crash with the 1.9.4 branch]

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2011-01-26T15:47:20Z
Modified: 2014-02-09T02:11:51Z
```

**Comment by taskwarrior** [comment]
_Bryce Harrington on 2011-01-26T15:48:54Z says:_ 

 Adding the task and then marking it done worked around the issue.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-01-26T15:54:17Z says:_ 

 Bryce, thanks for reporting this.

It is a known issue.  The problem is the combination of the "log" command, and specifying a project.  So instead of this:

<pre>
$ task log foo project:bar
</pre>

do this as a workaround:

<pre>
$ task add foo project:bar
$ task <id> done
</pre>

Sorry for the inconvenience.  Thanks for including the stack dump, most don't.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-01-29T17:33:52Z says:_ 

 Bryce, you said this:

<pre>
[Setting to priority high since it's a crash with the 1.9.4 branch]
</pre>

Could you confirm that please, because I believe it is fixed in 1.9.4.

**Comment by taskwarrior** [comment]
_Cory Donnelly on 2011-01-30T17:36:48Z says:_ 

 I don't experience this issue under OS X 10.4.11 on Intel.

**Comment by taskwarrior** [comment]
_Bryce Harrington on 2011-02-04T05:01:47Z says:_ 

 Paul Beckingham wrote:
> Bryce, you said this:
> 
> [...]
> 
> Could you confirm that please, because I believe it is fixed in 1.9.4.

Right, sorry I should have said this was against the 1.9.3 branch, not the 1.9.4 branch.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-02-04T10:48:08Z says:_ 

 thanks for checking Bryce.
I'm closing this issue - it was fixed in 1.9.4.
